packages/service-library/src/servicelib/rabbitmq/rpc_interfaces/webserver/functions/functions_rpc_interface.py

```diff
@@ -0,0 +1,325 @@
+        TypeAdapter(RPCMethodName).validate_python("delete_function"),
+        function_id=function_id,
+    )
+    assert result is None  # nosec
+    return result
+
+
+@log_decorator(_logger, level=logging.DEBUG)
+async def list_functions(
+    rabbitmq_rpc_client: RabbitMQRPCClient,
+    *,
+    pagination_offset: int,
+    pagination_limit: int,
+) -> tuple[list[RegisteredFunction], PageMetaInfoLimitOffset]:
+    result: tuple[list[RegisteredFunction], PageMetaInfoLimitOffset] = (
+        await rabbitmq_rpc_client.request(
+            WEBSERVER_RPC_NAMESPACE,
+            TypeAdapter(RPCMethodName).validate_python("list_functions"),
+            pagination_offset=pagination_offset,
+            pagination_limit=pagination_limit,
+        )
+    )
+    assert isinstance(result, tuple)
+    TypeAdapter(list[RegisteredFunction]).validate_python(
+        result[0]
+    )  # Validates the result as a list of RegisteredFunctions
+    assert isinstance(result[1], PageMetaInfoLimitOffset)  # nosec
+    return result
+
+
+@log_decorator(_logger, level=logging.DEBUG)
+async def list_function_jobs(
+    rabbitmq_rpc_client: RabbitMQRPCClient,
+    *,
+    pagination_limit: int,
+    pagination_offset: int,
+) -> tuple[list[RegisteredFunctionJob], PageMetaInfoLimitOffset]:
+    result: tuple[list[RegisteredFunctionJob], PageMetaInfoLimitOffset] = (
+        await rabbitmq_rpc_client.request(
+            WEBSERVER_RPC_NAMESPACE,
+            TypeAdapter(RPCMethodName).validate_python("list_function_jobs"),
+            pagination_offset=pagination_offset,
+            pagination_limit=pagination_limit,
+        )
+    )
+    assert isinstance(result, tuple)  # nosec
+    assert isinstance(result[0], list)  # nosec
+    assert all(
+        TypeAdapter(RegisteredFunctionJob).validate_python(item) for item in result[0]
+    )  # nosec
+    assert isinstance(result[1], PageMetaInfoLimitOffset)  # nosec
+    return (
+        TypeAdapter(list[RegisteredFunctionJob]).validate_python(result[0]),
+        TypeAdapter(PageMetaInfoLimitOffset).validate_python(result[1]),
+    )
+
+
+@log_decorator(_logger, level=logging.DEBUG)
+async def list_function_job_collections(
+    rabbitmq_rpc_client: RabbitMQRPCClient,
+    *,
+    pagination_limit: int,
+    pagination_offset: int,
+) -> tuple[list[RegisteredFunctionJobCollection], PageMetaInfoLimitOffset]:
+    result = await rabbitmq_rpc_client.request(
+        WEBSERVER_RPC_NAMESPACE,
+        TypeAdapter(RPCMethodName).validate_python("list_function_job_collections"),
+        pagination_offset=pagination_offset,
+        pagination_limit=pagination_limit,
+    )
+    assert isinstance(result, tuple)
+    return (
+        TypeAdapter(list[RegisteredFunctionJobCollection]).validate_python(
+            result[0]
+        ),  # Validates the result as a list of RegisteredFunctionJobCollections
+        TypeAdapter(PageMetaInfoLimitOffset).validate_python(result[1]),  # nosec
+    )
```

Comment:
and here

Reply:
done

Reply:
this one is still not vslidated as a tuple but as 2 different parts

Reply:
done